core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/locks/IOperationLockManager.java

```diff
@@ -17,65 +17,80 @@
+	 *            subsequent targets to lock (may not be {@code null}; individual elements may not be {@code null})
+	 * @throws LockedException
+	 *             when one or more locks for the given targets can not be acquired for some reason
 	 */
-	void lock(DatastoreLockContext context, long timeoutMillis, DatastoreLockTarget firstTarget, DatastoreLockTarget... restTargets) throws LockedException;
+	void lock(DatastoreLockContext context, long timeoutMillis, Lockable lockable, Lockable... lockables)
+			throws LockedException;
 
 	/**
 	 * Locks one or more {@link IOperationLockTarget targets} for the specified lock context.
 	 * 
-	 * @param context the lock context (may not be {@code null})
-	 * @param timeoutMillis the maximum allowed time in milliseconds in which this call may block (must be {@link #NO_TIMEOUT}, zero or positive)
-	 * @param targets the targets to lock (may not be {@code null}; can be empty)
-	 * @throws LockedException when one or more locks for the given targets can not be acquired for some reason
+	 * @param context
+	 *            the lock context (may not be {@code null})
+	 * @param timeoutMillis
+	 *            the maximum allowed time in milliseconds in which this call may block (must be {@link #NO_TIMEOUT}, zero or positive)
+	 * @param targets
+	 *            the targets to lock (may not be {@code null}; can be empty)
+	 * @throws LockedException
+	 *             when one or more locks for the given targets can not be acquired for some reason
 	 */
-	void lock(DatastoreLockContext context, long timeoutMillis, Iterable<DatastoreLockTarget> targets) throws LockedException;
+	void lock(DatastoreLockContext context, long timeoutMillis, Iterable<Lockable> targets) throws LockedException;
 
 	/**
 	 * Unlocks one or more {@link IOperationLockTarget targets} with the specified lock context.
 	 * 
-	 * @param context the lock context (may not be {@code null})
-	 * @param firstTarget the first (or only) target to unlock (may not be {@code null})
-	 * @param restTargets subsequent targets to unlock (may not be {@code null}; individual elements may not be {@code null})
-	 * @throws IllegalArgumentException when one or more locks for the given targets could not be released for some reason
+	 * @param context
+	 *            the lock context (may not be {@code null})
+	 * @param firstTarget
+	 *            the first (or only) target to unlock (may not be {@code null})
+	 * @param restTargets
+	 *            subsequent targets to unlock (may not be {@code null}; individual elements may not be {@code null})
+	 * @throws IllegalArgumentException
+	 *             when one or more locks for the given targets could not be released for some reason
 	 */
-	void unlock(DatastoreLockContext context, DatastoreLockTarget firstTarget, DatastoreLockTarget... restTargets) throws IllegalArgumentException;
+	void unlock(DatastoreLockContext context, Lockable firstTarget, Lockable... restTargets) throws IllegalArgumentException;
```

Comment:
The lock() counterpart has a `lockable` and a `lockables` parameter. What do you think about using something similar here, just for the sake of consistency? Or `target` and `targets` or whatever suits the purpose best. The `restTargets` tripped me to think about REST for some reason :dumbo:

Reply:
:nod: